main/api/src/mill/api/ClassLoader.scala

```diff
@@ -69,6 +69,19 @@ object ClassLoader {
         .getMethod("getPlatformClassLoader")
         .invoke(null)
         .asInstanceOf[ClassLoader]
+    } else {
+      // With Java 8 we want a clean classloader that still contains classes
+      // coming from com.sun.* etc.
+      // We get the application classloader parent which happens to be of
+      // type sun.misc.Launcher$ExtClassLoader
+      // We can't call the method directly since it doesn't exist in Java 9
+      // So we load it via reflection.
+      val launcherClass = getClass.getClassLoader().loadClass("sun.misc.Launcher")
+      val getLauncherMethod = launcherClass.getMethod("getLauncher")
+      val launcher = getLauncherMethod.invoke(null)
+      val getClassLoaderMethod = launcher.getClass().getMethod("getClassLoader")
+      val appClassLoader = getClassLoaderMethod.invoke(launcher).asInstanceOf[ClassLoader]
+      appClassLoader.getParent()
```

Comment:
Why do we load this code via reflection, when we are only supposed to run it on Java 8? It this more the kind of compile-time work-around?

Reply:
The code doesn't compile in Java 9+. I used reflection to allow compilation of Mill in Java 9+ since the `sun.misc` package doesn't exist in Java 9+.

Reply:
Thanks for clarifying. I think the comment is kind of abḿbigous. Maybe, you can add, that reflection solved an issue at compilation-time?

Reply:
Rephrased the commend a bit.